src/runtime_src/core/pcie/driver/linux/xocl/subdev/hwmon_sdm.c

```diff
@@ -0,0 +1,517 @@
+
+	switch(repo_type) {
+		case SDR_TYPE_BDINFO:
+			id = 0;
+			break;
+		case SDR_TYPE_TEMP:
+			id = 1;
+			break;
+		case SDR_TYPE_VOLTAGE:
+			id = 2;
+			break;
+		case SDR_TYPE_CURRENT:
+			id = 3;
+			break;
+		case SDR_TYPE_POWER:
+			id = 4;
+			break;
+		case SDR_TYPE_QSFP:
+			id = 5;
+			break;
+		case SDR_TYPE_VPD_PCIE:
+			id = 6;
+			break;
+		case SDR_TYPE_IPMIFRU:
+			id = 7;
+			break;
+		case SDR_TYPE_CSDR_LOGDATA:
+			id = 8;
+			break;
+		case SDR_TYPE_VMC_LOGDATA:
+			id = 9;
+			break;
+		default:
+			id = -1;
+			break;
+	}
+
+	return id;
+}
+
+static void parse_sdr_info(char *in_buf, struct xocl_hwmon_sdm *sdm,
+                           bool create_sysfs)
+{
+	bool create = false;
+	uint8_t status;
+	int buf_index, err;
+	uint8_t remaining_records, completion_code, repo_type, repo_id;
+	uint8_t name_length, name_type_length;
+	uint8_t val_len, value_type_length, threshold_support_byte;
+	uint8_t bu_len, sensor_id, base_unit_type_length, unit_modifier_byte;
+	uint32_t name_index, ins_index, max_index = 0, avg_index = 0;
+
+	completion_code = in_buf[SDR_COMPLETE_IDX];
+
+	xocl_dbg(&sdm->pdev->dev, "Parsing SDR Repository: received completion_code: 0x%x",
+			 completion_code);
+
+	if(completion_code != SDR_CODE_OP_SUCCESS)
+	{
+		if(completion_code == SDR_CODE_NOT_AVAILABLE)
+			xocl_err(&sdm->pdev->dev, "Error: SDR Code Not Available");
+		else if(completion_code == SDR_CODE_OP_FAILED)
+			xocl_err(&sdm->pdev->dev, "Error: SDR Code Operation Failed");
+		else if(completion_code == SDR_CODE_FLOW_CONTROL_READ_STALE)
+			xocl_err(&sdm->pdev->dev, "Error: SDR Code Flow Control Read Stale");
+		else if(completion_code == SDR_CODE_FLOW_CONTROL_WRITE_ERROR)
+			xocl_err(&sdm->pdev->dev, "Error: SDR Code Flow Control Write Error");
+		else if(completion_code == SDR_CODE_INVALID_SENSOR_ID)
+			xocl_err(&sdm->pdev->dev, "Error: SDR Code Invalid Sensor ID");
+		else
+			xocl_err(&sdm->pdev->dev, "Failed in sending SDR Repository command");
+		return;
+	}
+
+	repo_type = in_buf[SDR_REPO_IDX];
+	repo_id = sdr_get_id(repo_type);
+	if (repo_id < 0) {
+		xocl_err(&sdm->pdev->dev, "SDR Responce has INVALID REPO TYPE: %d", repo_type);
+		return;
+	}
+
+	buf_index = SDR_NUM_BYTES_IDX + 1;
+
+	remaining_records = in_buf[SDR_NUM_REC_IDX];
+
+	while(remaining_records > 0)
+	{
+		sensor_id = in_buf[buf_index++];
+
+		name_type_length = in_buf[buf_index++];
+		name_length = name_type_length & SDR_LENGTH_MASK;
+		name_index = buf_index;
+
+		buf_index += name_length;
+
+		value_type_length = in_buf[buf_index++];
+		val_len = value_type_length & SDR_LENGTH_MASK;
+		ins_index = buf_index;
+
+		buf_index += val_len;
+
+		base_unit_type_length = in_buf[buf_index++];
+		if(base_unit_type_length != SDR_NULL_BYTE)
+		{
+			bu_len = base_unit_type_length & SDR_LENGTH_MASK;
+			buf_index += bu_len;
+		}
+
+		unit_modifier_byte = in_buf[buf_index++];
+		threshold_support_byte = in_buf[buf_index++];
+
+		if(threshold_support_byte != SDR_NULL_BYTE)
```

Comment:
what if threshold_support_byte == SDR_NULL_BYTE?
the buf_index will not be += val_len?

Reply:
yes, it won't.